compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/RewriteFft.cpp

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//
+
+#include "iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.h"
+#include "iree/compiler/Dialect/LinalgExt/Transforms/Transforms.h"
+
+namespace mlir::iree_compiler::IREE::LinalgExt {
+
+FailureOr<std::pair<Value, Value>> rewriteFft(Operation *op, Value operand,
+                                              int64_t fftLength,
+                                              PatternRewriter &rewriter) {
+
+  assert(llvm::isPowerOf2_64(fftLength) &&
+         "expected FFT length to be a power of two");
+  Location loc = op->getLoc();
+
+  auto operandType = llvm::dyn_cast<RankedTensorType>(operand.getType());
+  if (!operandType || !operandType.hasStaticShape()) {
+    return failure();
+  }
+
+  // Skip else getBitReversalOrder produces invalid dense elements attr.
+  if (isa<ComplexType>(getElementTypeOrSelf(operand.getType())))
```

Comment:
Looking at tests it looks like the only supported element type is `f32`. We can strengthen this to
```suggestion
  if (!getElementTypeOrSelf(operand.getType()).isFloat32Type())
```

If other types are supported, can you add tests for them?

Reply:
True, I think the current implementation only supports `f32` and some more work would be needed to support other types (I would say outside the scope of this PR). I'll strengthen the check.